src/Analyzer/Resolve/evaluateScalarSubqueryIfNeeded.cpp

```diff
@@ -235,12 +236,12 @@ void QueryAnalyzer::evaluateScalarSubqueryIfNeeded(QueryTreeNodePtr & node, Iden
                 io.pipeline.setConcurrencyControl(context->getSettingsRef()[Setting::use_concurrency_control]);
 
                 executor.emplace(io.pipeline);
-                while (block.rows() == 0 && executor->pull(block))
+                while (chunk.getNumRows() == 0 && executor->pull(chunk))
```

Comment:
The fix is to replace Block with Chunk because an optimized for EXISTS plan may return a chunk with no columns but non-zero rows.